sleap/gui/web.py

```diff
@@ -146,6 +149,49 @@ def get_release(self, version: str) -> Release:
         )
 
 
+@attr.s(auto_attribs=True)
+class AnnouncementChecker:
+    """Checker for new announcements on the bulletin page of sleap."""
+
+    bulletin_json_path: str = BULLETIN_JSON
+    previous_announcement_date: str = prefs.prefs["announcement last seen date"]
```

Comment:
The `prefs.prefs` variable should only be read in at opening and written to at closing the program. For intermediate states, we use `MainWindow.state`. The `prefs.prefs` variable stores the user preferences which are written to a file to be remembered each time the GUI is opened/closed. Also, I don't think that this class would know what the `prefs` variable is if we ran it in some test code - maybe add both a fixture (for the bulletin.md) and a test that runs through all the methods in the announcement checker?

Reply:
Can we directly use `MainWindow.state["announcement last seen date"]` or do we have to use it in some other way?

Reply:
No, we would need to pass in the instance of `MainWindow` to access its `state` attribute. For example, this is how we initialize the `CommandContext`:
https://github.com/talmolab/sleap/blob/ed77b49164b654bf9e43c29fcc3af73be2f8eb3b/sleap/gui/app.py#L123-L125
. Here, `self` is the instance of `MainWindow`, and this bit of code resides in `MainWindow.__init__`.